### ISSUE TITLE
saving arrays in documents with plugins

### ISSUE THREAD

**Issue by dirkam** [baasbox/baasbox#692]
How can I save a document with arrays using plugins?

A document looks like this:

{
  "myid": "someid",
  "title": "sometitle",
  ...
  "category": [
    "cat1",
    "cat2",
    "cat3"
  ]
}

I try to update the document with the following code:
    Box.runAsAdmin(function(){
        Box.Documents.save("items",{
            "id":item.id,
            "myid":item.myid,
            "title":item.title,
            ...
            "category":item.category
        });
       });
    

Everything is saved correctly except the category, it simply disappears, there will be no "category" in the document.

**Comment by giastfader** [comment]
This works for me:

````
Javascript
/* uncomment this snippet to write a script that replies on GET method */
http().get(function(req){
    var item={
        "myid": "someid",
        "title": "sometitle",
        "category": [
        "cat1",
        "cat2",
        "cat3"
        ]
    };

    Box.runAsAdmin(function(){
        Box.Documents.save("items",{
            "myid":item.myid,
            "title":item.title,
            "category":item.category
        });
    });
      
});
````
![screen shot 2015-03-02 at 13 34 08](https://cloud.githubusercontent.com/assets/2334584/6440774/e8c8de74-c0e0-11e4-821b-7b670b401ae2.png)

I see that you set also the "id" field, in this case BaasBox tries to update the existing document rather than to create a new one.
Can be this the problem?

**Comment by dirkam** [comment]
Yes, I try to update the document, so that might be the issue. I create the documents by adding them on the admin console and it works fine.

By the way is there any way to only update some fields in a document? Based on my experience I have to specify every field when updating otherwise it will be removed.

**Comment by giastfader** [comment]
Not via plugin.
This feature is available via REST API:
http://www.baasbox.com/documentation/?shell#update-a-document-39-s-field

**Comment by dirkam** [comment]
I see. This is why I try to use the code above. Could you please try the update as well?
Btw I'm running 0.9.1.

**Comment by giastfader** [comment]
Done, it works:

```Javascript
/* uncomment this snippet to write a script that replies on GET method */
http().get(function(req){
    var item={
        "id":"86c4b404-c5cf-4d63-bbc3-e31d668dbc95",
        "myid": "someid2",
        "title": "sometitle",
        "category": [
            "cat1",
            "cat2",
            "cat3"
        ]
    };
    Box.runAsAdmin(function(){
        Box.Documents.save("items",{
            "id":item.id,
            "myid":item.myid,
            "title":item.title,
            "category":item.category
        });
    });
});
```

**Comment by dirkam** [comment]
It's strange, I double checked it and for me it doesn't work. If I change the category to a simple string value the same code updates it fine, however, in case of an array value the category field is missing after the update.

**Comment by giastfader** [comment]
does my code work in your BaasBox?

**Comment by dirkam** [comment]
It does.

This one doesn't:

    function getItems(){
        var value = Box.runAsAdmin(function(){
        var ret=Box.Documents.find("items");
        return  ret;
    });
    return value;
    }

   var items = getItems();
    for (i = 0; i < items.length; i++){
        updateItem(items[i]);
    }

    function updateItem(item){
        Box.log(item.category);
    Box.runAsAdmin(function(){
    Box.Documents.save("items",{
            "id":item.id,
            ...
            "category":item.category
        });
        });
    }

**Comment by giastfader** [comment]
Ok, your code does not work!
This is weird :\

**Comment by avently** [comment]
Yes, guys. I have a problems with arrays too. I thought it's my bad...

**Comment by giastfader** [comment]
It only happens if the array has been read from the database. It is a Java/Javascript bridge problem. We are working on it.
